apps/ui/src/main.tsx

```diff
@@ -0,0 +1,10 @@
+import { StrictMode } from 'react'
+import { createRoot } from 'react-dom/client'
+import App from './App.tsx'
+import './index.css'
+
+createRoot(document.getElementById('root')!).render(
```

Comment:
what does the `!` do here?

Reply:
insists this will be not null. its a shortcut around TS type checking for either "I know better" or "Its ok for things to break here". This is becoming a pretty standard setup for React apps since if this node is missing the whole website will fail to render and so there isn't really a fallback.